nvflare/tool/job/config/configer.py

```diff
@@ -177,33 +186,44 @@ def merge_configs(
                                         raise ValueError(f"invalid key {key} for file {file}")
-
+                                key_value = None
                                 parent, index, key = split_array_key(key)
                                 if parent is not None and index is not None:
                                     parent_config_list = config.get(parent)
                                     if not isinstance(parent_config_list, list):
                                         raise ValueError(f"invalid key '{key}' for file {file}")
                                     index_config = parent_config_list[index]
-
-                                    index_config.put(key, cli_value)
-                                    key_value = index_config.get(key)
+                                    if cli_value:
+                                        index_config.put(key, cli_value)
+                                        key_value = index_config.get(key)
+                                    else:
+                                        index_config.pop(key)
                                 else:
-                                    config.put(key, cli_value)
-                                    key_value = config.get(key)
-
-                                tokens = key.split(".")
-                                if len(tokens) > 1:
-                                    key = tokens[-1]
-                                key_index = KeyIndex(key, key_value)
-                                key_indices[key] = [key_index]
+                                    if cli_value:
+                                        config.put(key, cli_value)
+                                        key_value = config.get(key)
+                                    else:
+                                        config.pop(key)
+                                if key_value:
+                                    tokens = key.split(".")
+                                    if len(tokens) > 1:
+                                        key = tokens[-1]
+                                    key_index = KeyIndex(key, key_value)
+                                    key_indices[key] = [key_index]
+                                else:
+                                    last_token = get_last_token(key)
+                                    key_indices.pop(last_token)
                             else:
-                                indices = key_indices.get(key)
-                                for key_index in indices:
-                                    new_value = _cast_type(key_index, cli_value)
-                                    key_index.value = new_value
-                                    parent_key = key_index.parent_key
-                                    if parent_key and isinstance(parent_key.value, ConfigTree):
-                                        parent_key.value.put(key_index.key, new_value)
+                                if cli_value:
```

Comment:
This else is really far away from the if statement.  We should think about how to make it easier to read.